### PR TITLE
php.packages.{box,castor}: build from source

### PR DESCRIPTION
**Pull request by etu**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by drupol — APPROVED** [review]
<3

**Comment by drupol** [comment]
By the way, how about adding `mainProgram` everywhere?

**Review by drupol — DISMISSED** [review]

**Comment by etu** [comment]
Seems like I broke the castor completions, I'll look into it later

**Comment by gaelreyrol** [comment]
> Seems like I broke the castor completions, I'll look into it later

That's what I wanted to ask, you remove the small part to init castor before running the install shell completion part.

**Comment by etu** [comment]
> > Seems like I broke the castor completions, I'll look into it later
> 
> That's what I wanted to ask, you removed the small part to init castor before running the install shell completion part.

I also tried to add just that line back and it still didn't work. So I have to investigate further.

File: pkgs/development/php-packages/grumphp/default.nix [hunk 1]
```diff
@@ -1,32 +1,24 @@
...
-    sha256 = "sha256-EqzJb7DYZb7PnebErLVI/EZLxj0m26cniZlsu1feif0=";
+  src = fetchFromGitHub {
+    owner = "phpro";
+    repo = "grumphp";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-gQs05nrKh1A6N6xmoPayqoqFZdV2zK/2UAg6ZkpDzBs=";
   };
 
-  dontUnpack = true;
+  vendorHash = "sha256-iaLhs1lKo4X2ix6WjtgJDKc2NiPQpOlwrSR8ois6aaM=";
```
Comment (drupol) [inline pkgs/development/php-packages/grumphp/default.nix:14]: The attribute `composerLock` is missing.

Context: https://github.com/phpro/grumphp/issues/1095

Possible solution:

```
  composerLock = fetchurl {
    url = "https://github.com/phpro/grumphp-shim/raw/695f2f0be5f2790242c362e8b79d3f92c276c9a1/phar.composer.lock";
    hash = "sha256-XeWDUQUnUEOCtT+wjmR0kXikGHdE336KXglTfzfVVQo=";
  };
```

**Comment by etu** [comment]
Castor has completions now!

File: pkgs/development/php-packages/grumphp/default.nix (hunk 1 above)
Comment (etu) [reply pkgs/development/php-packages/grumphp/default.nix]: Well, it does build for me from this commit that is pushed. I tested adding the following `installPhase`: `ls; exit 1` and there's a `composer.lock` in there. So I guess they have added it to the release tarball?

**Comment by etu** [comment]
@ofborg build php81.packages.box php81.packages.castor php81.packages.grumphp php82.packages.box php82.packages.castor php82.packages.grumphp php83.packages.box php83.packages.castor php83.packages.grumphp

File: pkgs/development/php-packages/grumphp/default.nix (hunk 1 above)
Comment (etu) [reply pkgs/development/php-packages/grumphp/default.nix]: Not sure how I got it to build once. I'm dropping the changes for that package for now.

**Comment by etu** [comment]
https://github.com/ofborg build php81.packages.box php81.packages.castor php82.packages.box php82.packages.castor php83.packages.box php83.packages.castor

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
@GrahamcOfBorg build box castor

**Pull request merged**